File: Core/Entities/TestResultDetail.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Entities
{
    public class TestResultDetail
    {
        public int Id { get; set; }

        [Required]
        public string Uri { get; set; }

        public long ResponseTime { get; set; }

        public long MinResponseTime { get; set; }
        public long MaxResponseTime { get; set; }
        public double MeanResponseTime { get; set; }

        public TestResult TestResult { get; set; }
    }
}

File: angular/src/app/models/test-result-detail.ts
export class TestResultDetail {
    id: number;
    uri: string;
    responseTime: number;
    minResponseTime: number;
    maxResponseTime: number;
    meanResponseTime: number;
    resultId: number;

    constructor(obj: any) {
        const { id, uri, responseTime, minResponseTime, maxResponseTime, meanResponseTime, testResultId  } = obj;
        const { Id, Uri, ResponseTime, MinResponseTime, MaxResponseTime, MeanResponseTime, TestResultId  } = obj;

        this.id = id || Id;
        this.uri = uri || Uri;
        this.responseTime = responseTime || ResponseTime;
        this.minResponseTime = minResponseTime || MinResponseTime;
        this.maxResponseTime =  maxResponseTime || MaxResponseTime;
        this.meanResponseTime = meanResponseTime || MeanResponseTime;
        this.resultId = testResultId || TestResultId;
    }
}
